### PR TITLE
fix replacement bug

### PR DESCRIPTION
**Pull request by fenekku**
- update README.md for v0.3
- fix replacement bug not accounting for relabelling:
  MeSH replacements are given in terms of old label to new label. But sometimes the new label is to a renamed(relabelled) existing subjects. That wasn't taken into account before which led to additions/removals where replacements should have happened. 
- :package: release: v0.3.2

### CODE REVIEW

File: tests/contrib/mesh/test_converter.py [hunk 1]
```diff
@@ -86,15 +86,32 @@ def subject_qualified_present_only_in_dst():
     return Subject(id="D005654Q000187", label="Fundus Oculi/drug effects")
 
 
-def renamed_subjects():
+def subjects_renamed():
```
Comment (fenekku) [inline tests/contrib/mesh/test_converter.py:89]: Did some renaming for clarity.

File: tests/contrib/mesh/test_converter.py [hunk 2]
```diff
@@ -186,55 +221,74 @@ def test_converter_rename():
...
+        subject_rename_src,
     ]
     dst = [
         subject_unqualified(),
-        replacement_subject_1,
-        replacement_subject_2,
+        subject_replace_present_dst,
+        subject_replace_add_dst,
+        subject_replace_rename_dst,
+        subject_rename_dst,
```
Comment (fenekku) [inline tests/contrib/mesh/test_converter.py:259]: I am still not terribly happy with the clarity of the code here, but best I could come up with.

File: galter_subjects_utils/contrib/mesh/converter.py [hunk 3]
```diff
@@ -143,46 +143,60 @@ def _analyze_dst(self):
...
+
+            # is replacement_label for newly added subject?
+            replacement_subject = next(
+                (s for s in self._additions if s.label == replacement_label),
+                None
+            )
+            if replacement_subject:
+                return replacement_subject.id
+
+            # is replacement_label for relabelled existing and kept subject?
```
Comment (fenekku) [inline galter_subjects_utils/contrib/mesh/converter.py:180]: This is the case that wasn't taken into account before. The rest is refactoring for clarity.

File: tests/contrib/mesh/test_converter.py (hunk 2 above)
Comment (mattbaumann1) [reply tests/contrib/mesh/test_converter.py:259]: I think this is more clear. Thank you.

**Review by mattbaumann1 — APPROVED** [review]

**Review by carsonicator — APPROVED** [review]
Thanks @fenekku

File: galter_subjects_utils/contrib/mesh/converter.py [hunk 4]
```diff
@@ -143,46 +143,60 @@ def _analyze_dst(self):
...
+            if not replacement_label:
+                return None
+
+            # is replacement_label for existing and kept subject?
+            replacement_analysis = self._label_to_analysis.get(replacement_label)  # noqa
+            if replacement_analysis and replacement_analysis.seen:
+                return replacement_analysis.id
+
+            # is replacement_label for newly added subject?
+            replacement_subject = next(
```
Comment (Meowcenary) [inline galter_subjects_utils/contrib/mesh/converter.py:173]: The `next` function always takes me a second to grasp. It's pythonic, I'm just not used to it still.

**Comment by Meowcenary** [comment]
Still reviewing this. Almost done

File: tests/contrib/mesh/test_converter.py [hunk 5]
```diff
@@ -128,6 +145,24 @@ def replaced_subjects_newly_added():
     )
 
 
+def subjects_replaced_renamed():
+    """Return a subject and its replaced version for the relabelled scenario.
+
+    This is to be used for the case where a subject is replaced by
```
Comment (Meowcenary) [inline tests/contrib/mesh/test_converter.py:151]: 👍 This makes it clear what the edge case was.

File: tests/contrib/mesh/test_converter.py (hunk 2 above)
Comment (Meowcenary) [reply tests/contrib/mesh/test_converter.py:259]: Eh, sometimes there's not much you can do 🤷 I think with a bit of reading someone can reasonably jump in and figure it out and that's about as good as it gets.

**Review by Meowcenary — APPROVED** [review]
It always takes me a bit to wrap my head around these, but it seems good.

File: galter_subjects_utils/contrib/mesh/converter.py (hunk 4 above)
Comment (fenekku) [reply galter_subjects_utils/contrib/mesh/converter.py:173]: Yes, it's a somewhat round-about way of implementing "find first instance in array satisyfing these conditions". It's more compact than a for-loop for that purpose and more efficient.
